mmv1/api/product.go

```diff
@@ -240,7 +240,8 @@ func Merge(self, otherObj reflect.Value) {
 
 		// skip if the override is the "empty" value
 		emptyOverrideValue := reflect.DeepEqual(reflect.Zero(otherObj.Field(i).Type()).Interface(), otherObj.Field(i).Interface())
-		if emptyOverrideValue {
+
+		if emptyOverrideValue && selfObj.Type().Field(i).Name != "Required" {
```

Comment:
One of the private overrides actually has `Required: false`, which means we can't just treat it as an empty boolean value.

Reply:
Yea, I remember this one. With the ruby implementation, you couldn't otherwise change a required field to optional, so yea, we'd need to keep this unless the Go version handles it differently.